### PR TITLE
refactor(zoe): Replace arrayToObj with objectMap

### PR DESCRIPTION
**Pull request by gibson042**
## Description

`arrayToObj` is an old function and has significant overlap with `objectMap` (the latter is basically `Object.fromEntries(Object.entries(obj).map(([key, value]) => [key, mapFn(value, key)]))`, and the former is basically `Object.fromEntries(zip(keys, values))` where values comes from manually mapping over original values).

This PR removes it in favor of more idiomatic code.

### Security Considerations

n/a

### Scaling Considerations

n/a

### Documentation Considerations

Probably none; no new code is using the affected functions.

### Testing Considerations

All remaining tests should continue to pass.

### Upgrade Considerations

n/a; behavior is unaffected

### CODE REVIEW

**Review by erights — COMMENTED** [review]
I'll wait until this is ready for review before doing a more careful read. But looks good to me so far. A welcome simplification!

File: packages/zoe/src/issuerStorage.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
+import { deeplyFulfilledObject, objectMap } from '@agoric/internal';
```
Comment (erights) [inline packages/zoe/src/issuerStorage.js:1]:
```suggestion
import { objectMap } from '@endo/common/object-map.js';
import { deeplyFulfilledObject } from '@agoric/internal';
```

And all other occurrences.

**Review by erights — CHANGES_REQUESTED** [review]
Please do fix all the imports of this. Thanks.

**Comment by gibson042** [comment]
There is a test failure in dapp-card-store integration caused by promise chains internal to `deeplyFulfilled` via `E.when`. It should be resolved by https://github.com/Agoric/dapp-card-store/pull/73 .

File: packages/zoe/src/issuerStorage.js (hunk 1 above)
Comment (gibson042) [reply packages/zoe/src/issuerStorage.js:1]: I see this in 16 other files; can we defer it to a dedicated PR?
```console
$ git grep -l "import.*objectMap.* from '@agoric/internal'" packages/ | tee /dev/stderr | wc -l
packages/SwingSet/test/upgrade/test-upgrade.js
packages/SwingSet/tools/bootstrap-relay.js
packages/agoric-cli/src/commands/inter.js
packages/builders/scripts/inter-protocol/init-core.js
packages/governance/src/contractHelper.js
packages/inter-protocol/test/auction/test-scheduler.js
packages/inter-protocol/test/auction/tools.js
packages/inter-protocol/test/vaultFactory/driver.js
packages/inter-protocol/test/vaultFactory/test-replacePriceAuthority.js
packages/inter-protocol/test/vaultFactory/test-vaultFactory.js
packages/inter-protocol/test/vaultFactory/test-vaultLiquidation.js
packages/smart-wallet/src/offerWatcher.js
packages/smart-wallet/src/utils.js
packages/wallet/api/src/lib-wallet.js
packages/zoe/src/cleanProposal.js
packages/zoe/src/contractFacet/zcfZygote.js
packages/zoe/src/contractSupport/durability.js
packages/zoe/src/issuerStorage.js
packages/zoe/src/zoeService/escrowStorage.js
19
```

Comment (erights) [reply packages/zoe/src/issuerStorage.js:1]: Sigh. Yes.

Comment (erights) [reply packages/zoe/src/issuerStorage.js:1]: Resolving.

**Review by erights — APPROVED** [review]
LGTM

**Comment by gibson042** [comment]
@erights: I added a new commit relating to yesterday's discussion: https://github.com/Agoric/agoric-sdk/pull/8856/commits/726c44d5c8fd6d4e58ff8cab86de384c86205ed6

**Review by kriskowal — APPROVED** [review]
Looks mechanical to me.

**Comment by erights** [comment]
> @erights: I added a new commit relating to yesterday's discussion: [726c44d](https://github.com/Agoric/agoric-sdk/commit/726c44d5c8fd6d4e58ff8cab86de384c86205ed6)

The disjointness of keywords will remain. So under that assumption, this indeed has no observable change. But it is more robust to possible future design changes as well as lack-of-enforcement errors.

Re "yesterday's discussion", I think that had to do with a restriction on having the "same" asset in give and want, for some def of "same" I cannot recall. In any case, @Chris-Hibbert is not finding that so my memory is likely confused. In any case, the problem we wanted to solve by lifting this (possibly non-existent) check would not be addressed by relaxing the disjoint keyword prohibition. They're independent.

In any case, still LGTM. Please proceed.

**Comment by Chris-Hibbert** [comment]
Most of this is a clean refactor. The code added in https://github.com/Agoric/agoric-sdk/commit/726c44d5c8fd6d4e58ff8cab86de384c86205ed6 looks like a semantic change. I'm going to add it to my list to be included in upgrade 16.

**Pull request merged**